### PR TITLE
Add typing for MFA

### PR DESCRIPTION
**Pull request by mattgd**
## Description
Add typing for MFA.

## Documentation

Does this require changes to the WorkOS Docs? E.g. the [API Reference](https://workos.com/docs/reference) or code snippets need updates.

```
[ ] Yes
```

If yes, link a related docs PR and add a docs maintainer as a reviewer. Their approval is required.

### CODE REVIEW

File: workos/utils/request.py [hunk 1]
```diff
@@ -48,7 +48,8 @@ def set_base_api_url(self, base_api_url):
     def generate_api_url(self, path):
         return self.base_api_url.format(path)
 
-    def build_parameterized_url(self, url, **params):
+    @classmethod
+    def build_parameterized_url(cls, url, **params):
```
Comment (mattgd) [inline workos/utils/request.py:52]: Still on the fence about the proper home for this method. Waiting until we're no longer RequestHelper dependent to move around.

File: workos/mfa.py [hunk 2]
```diff
@@ -78,18 +76,7 @@ def enroll_factor(
...
-        if type not in ["sms", "totp"]:
-            raise ValueError("Type parameter must be either 'sms' or 'totp'")
-
-        if (
-            type == "totp"
-            and totp_issuer is None
-            or type == "totp"
-            and totp_user is None
-        ):
+        if type == "totp" and (totp_issuer is None or totp_user is None):
```
Comment (tribble) [inline workos/mfa.py:79]: @PaulAsjes food for thought: it's hard to enforce these sorts of `if 'totp' then require `issuer` and `user` constraints clearly in a rest API. But we have more options in an SDK.

If we were rewriting this SDK from scratch, would it be better to have separate functions `enroll_sms` and `enroll_totp` with different required params?

**Review by tribble — APPROVED** [review]
Ship it.

weirdo gif

File: workos/mfa.py (hunk 2 above)
Comment (mattgd) [reply workos/mfa.py:79]: +1 to this. I thought about having to separate methods, or better yet overloaded methods.

**Pull request merged**